### ISSUE TITLE
Improve the background placeholder of the spacer block

### ISSUE THREAD

**Issue by youknowriad** [WordPress/gutenberg#28092]
When you select a spacer block, it gets a white/light background to be able to distinguish it. This is not great in context where there's a background image behind it. For instance:

 - Imagine there's a block pattern with a spacer block inside a Cover block 
 - A user inserting the cover block won't know that there's a spacer block since it's invisible
 - When trying to select the cover block, he might actually select the spacer block which is fine but I believe we can maybe improve things a bit if we had a semi-transparent background or something like that to clarify it (or maybe some kind of  shared pattern instead of a solid background).

cc @jasmussen what do you think?

**Comment by jasmussen** [comment]
I could swear I tried this in a PR and closed it, but that has to have been a local branch I never pushed. In other words, I had the same thought and generally agree with the idea. But when I tried it, I found that I couldn't get it to work well. If I recall correctly, I tried to keep the general gray color it has now, only semitransparent, which meant a black transparent color in light themes, and a white transparent color in dark themes. 

And it worked well in basic examples, but always fell apart, somehow. For example, let's say you use TwentyTwentyOne, a light theme — that would mean a black semitransparent color for the spacer. Now you insert a cover block with a black background, with a spacer inside. The spacer will be wholly invisible because the the semitransparent black disappears on the black background.

It's not a unique problem, we have it already with the select and focus styles, which can disappear if you have an unfortunate clash of colors. But this is mitigated by using two colors:

<img width="474" alt="Screenshot 2021-01-11 at 13 16 25" src="https://user-images.githubusercontent.com/1204802/104181569-53141600-540f-11eb-949e-78dca5970505.png">

<img width="542" alt="Screenshot 2021-01-11 at 13 16 43" src="https://user-images.githubusercontent.com/1204802/104181573-54454300-540f-11eb-8c52-05980ac5e38b.png">

Figma does the same. Here I've been evil and chosen the same blue color for two boxes, that the Figma select style has:

<img width="986" alt="Screenshot 2021-01-11 at 13 14 03" src="https://user-images.githubusercontent.com/1204802/104181593-60c99b80-540f-11eb-8a28-d39ae844fef6.png">

The white handles remain, but the border disappears. 

That's kind of the extent of it — unless we can find a color we _know_ will always contrast the background, we should use single colors sparingly, knowing the tradeoff, and ideally avoid opacities. (I even tried things like `mix-blend-mode`, but there's always a case where that fails too — [here's an example](https://turtleshell.com/5), the dots in the bottom left have plenty of contrast, the dots at the center right do not.)

So I definitely think we can find a better design for the spacer block, but I don't think opacity alone is going to work. What do other services do? Is there any precedence we can take inspiration from?
